web/src/user_events.ts

```diff
@@ -113,12 +114,16 @@ export const update_person = function update(event: UserUpdate): void {
             user.is_admin || event.role === settings_config.user_role_values.moderator.code;
         settings_users.update_user_data(event.user_id, event);
         user_profile.update_profile_modal_ui(user, event);
+        settings_account.set_user_own_role_dropdown_value();
+        settings_account.add_or_remove_owner_from_role_select();
+        settings_account.update_user_own_role_select_state();
```

Comment:
I think only the function to set dropdown value should be called for all cases. If not can you specify the scenario? As per my thinking, others are only needed for certain cases like removing "Owner" option is only needed if `is_owner` is changed and updating dropdown disabled state and tooltip is needed as explained [here](https://github.com/zulip/zulip/pull/34861/files#r2168870364).